src/digraph-cyclicity.jl

```diff
@@ -0,0 +1,439 @@
+abstract Visitor
+
+"""
+```transitiveclosure!(dg::DiGraph, selflooped = false)```
+
+Compute the transitive closure of a directed graph, using the Floyd-Warshall
+algorithm.
+
+Version of the function that modifies the original graph.
+
+Note: This is an O(V^3) algorithm. 
+
+# Arguments
+* `dg`: the directed graph on which the transitive closure is computed.
+* `selflooped`: whether self loop should be added to the directed graph,
+default to `false`.
+"""
+function transitiveclosure!(dg::DiGraph, selflooped = false)
```

Comment:
I wrote it that way because selfloop is tested only once, whereas otherwise it is tested at each iteration. But I don't really know deeply how the Julia's compilation works